scipy/stats/_discrepancy.pyx

```diff
@@ -0,0 +1,502 @@
+# cython: language_level=3
+# cython: boundscheck=False
+# cython: wraparound=False
+# cython: cdivision=True
+
+# distutils: language = c++
+
+import numpy as np
+cimport numpy as np
+from libc.math cimport fabs, sqrt, pow
+
+np.import_array()
+
+cdef extern from "<thread>" namespace "std" nogil:
+    cdef cppclass thread:
+        thread()
+        void thread[A, B, C, D, E, F](A, B, C, D, E, F)
+        void join()
+
+cdef extern from "<mutex>" namespace "std" nogil:
+    cdef cppclass mutex:
+        void lock()
+        void unlock()
+
+cdef extern from "<functional>" namespace "std" nogil:
+    cdef cppclass reference_wrapper[T]:
+        pass
+    cdef reference_wrapper[T] ref[T](T&)
+
+from libcpp.vector cimport vector
+
+
+cdef mutex THREADED_SUM_MUTEX
```

Comment:
in C/C++, uppercase names are used for constants. I am not sure using a uppercase name is a good idea for the mutex.

Reply:
Resolved in 6e1354e97c0be68bdb4f35cb534875c22071c8cc